src/globalComponents/apiCalls/marketPlaceAPIConnections.js

```diff
@@ -134,10 +134,83 @@ function marketSale(userId: string, sellerId: string, saleId: string, onLoad:()
 	);
 }
 
+function getUsersCurrentSales(userId:string, onLoad:(currentListings: Array<SaleObject>) => void): void {
+	const responsePromise: Promise<Response> = fetch('/api/marketplace/getUsersMarketSales/' + userId, {
+		method: 'get',
+		headers: {
+			'Access-Control-Allow-Origin': '*',
+			'Content-Type': 'application/json',
+			'Access-Control-Allow-Credentials': 'true',
+		},
+	});
+	responsePromise.then(
+		response => response.json().then(data => {
+			if (response.status !== 200) {
+				console.log(response.status);
+				console.log(data.msg);
+				console.log(data);
+			}
+			else {
+				const currentListings: Array<SaleObject> = [];
+				for (const sale of data) {
+					if(sale.itemType === "component") {
+						currentListings.push(new SaleObject(
+							sale.itemId,
+							sale.salePrice,
+							sale.amount,
+							sale.sellerId,
+							sale._id
+						));
+					}
+					else if(sale.itemType === "tank") {
+						currentListings.push(new SaleObject(
+							sale.itemId.tankName,
+							sale.salePrice,
+							sale.amount,
+							sale.sellerId,
+							sale._id
+						));
+					}
+					
+				} 
+				onLoad(currentListings);
+			}
+		})
+	).catch(
+		error => {
+			toast.error('Couldnt connect to server!');
+			console.log(error);
+		}
+	);
+}
+
+function removeASale(saleId: string): void {
+	const responsePromise: Promise<Response> = fetch('/api/marketplace/removeAMarketSale/', {
+		method: 'delete',
+		headers: {
+			'Access-Control-Allow-Origin': '*',
+			'Content-Type': 'application/json',
+			'Access-Control-Allow-Credentials': 'true',
+		},
+		body: JSON.stringify({ saleId:saleId }),
+	});
+	responsePromise.then(
+		response => response.json().then(data => {
+			toast.error(data.msg);
```

Comment:
You aren't checking for what response status is being returned here. You need an if statement that checks if the response.status !== 201. If it doesn't console.log the response.status, the data, and then do toast.error(getErrorFromObject(data)). Otherwise, if it is 201, then you can do a toast.success('Removed Sale!') (not toast.error).